macaroons/store.go

```diff
@@ -332,6 +332,30 @@ func (r *RootKeyStorage) GenerateNewRootKey() error {
 	}, func() {})
 }
 
+// SetRootKey sets the default macaroon root key, replacing the previous root
+// key if it existed.
+func (r *RootKeyStorage) SetRootKey(rootKey []byte) error {
+	if r.encKey == nil {
+		return ErrStoreLocked
+	}
+	if len(rootKey) != 32 {
+		return fmt.Errorf("root key must be 32 bytes")
+	}
+
+	return kvdb.Update(r.Backend, func(tx kvdb.RwTx) error {
+		bucket := tx.ReadWriteBucket(rootKeyBucketName)
+		if bucket == nil {
+			return ErrRootKeyBucketNotFound
+		}
+
+		encryptedKey, err := r.encKey.Encrypt(rootKey)
```

Comment:
Encrypt root key outside of DB transaction to block it for a shorter amount of time?